doc/nrf/libraries/others/emds.rst

```diff
@@ -14,43 +14,55 @@ Implementation
+The partition storage space must be larger than the combined data stored by the application.
 Allocating a larger storage area will demand more resources, but also increase the life expectancy of the persistent memory.
 The chosen size should reflect the amount of data stored, the available persistent memory and how the application calls the :c:func:`emds_store` function.
-In general, it should not be necessary to allocate a large storage area, since only a limited set of data should be stored to ensure swift completion of writing the data on shutdown.
 
 The memory location that is going to be stored must be added on initialization.
 All memory areas must be provided through entries containing an ID, data pointer, and data length, using the :c:func:`emds_entry_add` function and the :c:macro:`EMDS_STATIC_ENTRY_DEFINE` macro.
 Entries to be stored when the emergency data storage is triggered need their own unique IDs that are not changed after a reboot.
-
-When all entries are added, the :c:func:`emds_load` function restores the entries into the memory areas from the persistent memory.
+The entries are stored in the partition in form of snapshots.
+The snapshot is a packed copy of all entries that are registered in the EMDS stored in the data area of the partition.
+Additionally, each snapshot is described by the snapshot metadata that is stored in the metadata area of the partition.
+
+When all entries are added, the :c:func:`emds_load` function starts a partition scanning procedure to find the latest snapshot.
+The EMDS scans the metadata area of both partitions.
+The :kconfig:option:`CONFIG_EMDS_SCANNING_FAILURES` Kconfig option configures the depth of the scanning procedure.
+If the scanning procedure fails the configured number of times in a row, the scanning is stopped, because most likely, there is no stored metadata left.
+The scanning procedure gathers the most recent snapshots (set by :kconfig:option:`CONFIG_EMDS_MAX_CANDIDATES`) from the partition metadata area.
+After the scanning procedure is completed, the EMDS checks data integrity on the gathered candidates, starting from the latest one stored on the partition.
+If the data integrity checks pass, the entries are restored into the memory areas from the persistent memory.
+If candidates are not found or the data integrity check does not pass, the function returns an error.
+The :c:func:`emds_load` function returns an error code if it runs on the empty partitions.
 
 After restoring the previous data, the application must run the :c:func:`emds_prepare` function to prepare the storage area for receiving new entries.
-If the remaining empty storage area is smaller than the required data size, the storage area will be automatically erased to increase the available storage area.
+Initially, the EMDS tries to allocate the storage area for the new entries in the partition that has the most recent snapshot to use the partition capacity at maximum.
```

Comment:
minor: optional:

```suggestion
Initially, the EMDS library tries to allocate the storage area for the new entries in the partition that has the most recent snapshot in order to use the full partition capacity.
```
I think similar updates are needed at other places as well. Because sometimes this doc uses "EMDS" to mean storage area, sometimes it uses "EMDS" to mean library, and sometimes to mean a specific mechanism. That causes a bit of confusion when someone is reading this for the first time.